### ISSUE TITLE
OpenAPI Spec Rules wildcard "*" is not accepted as a path param. 

### ISSUE THREAD

**Issue by DanCRichards** [stoplightio/spectral#2647]
**Describe the bug**
The spectral:oas ruleset does not allow for {*} to be included within the path params. 

**To Reproduce**

1. Consider the following endpoint which has been defined by fastify-swagger. 
```yaml
openapi: 3.0.3
info:
  title: "Wildcard Issue API"
  version: 4.2.0
  description: :) 
components:
  schemas: {}
paths:
  /documentation/static/{*}:
    head: &a1
      parameters:
        - schema:
            type: string
          in: path
          name: "*"
          required: true
      responses:
        "200":
          description: Default Response
    get: *a1

```
2. Run this CLI command `npx spectral lint openapi.yml -D` 
3. See error `error path-params Parameter "*" must be used in path "/documentation/static/{*}". paths./documentation/something/{*}.head.parameters[0]`
4. 

**Expected behavior**
I would expect that {*} is an accepted path param 
**Screenshots**
![image](https://github.com/stoplightio/spectral/assets/17732740/b12c04e6-0534-4581-9b81-66ca2ea676c8)

**Problematic Code** 
All of this code is in oasPathParam.js

Code raising error. [here](https://github.com/stoplightio/spectral/blob/048924d358ccb6432e368ae3d0b15cb78c72a6dd/packages/rulesets/src/oas/functions/oasPathParam.ts#L46)
```javascript
const ensureAllDefinedPathParamsAreUsedInPath = (path, params, expected, results) => {
    for (const p of Object.keys(params)) {
        if (!params[p]) {
            continue;
        }
        if (!expected.includes(p)) {
            const resPath = params[p];
            results.push(generateResult(`Parameter "${p}" must be used in path "${path}".`, resPath));
        }
    }
};
```

In the function above the expected argument is empty for the endpoint in question. The function is called by oasPathParams(path) function [here](https://github.com/stoplightio/spectral/blob/048924d358ccb6432e368ae3d0b15cb78c72a6dd/packages/rulesets/src/oas/functions/oasPathParam.ts#L87)

```javascript
function oasPathParam(paths) {
         /*
            ... 
         */

        const pathElements = [];
        let match;
        while ((match = pathRegex.exec(path))) {
            const p = match[0].replace(/[{}?*;]/g, '');
            if (pathElements.includes(p)) {
                results.push(generateResult(`Path "${path}" must not use parameter "{${p}}" multiple times.`, ['paths', path]));
            }
            else {
                pathElements.push(p);
            }
        }

       /*
          ...
      */

       ensureAllExpectedParamsInPathAreDefined(path, definedParams, pathElements, operationPath, results);
}
```

When running the endpoint path in question isn't validated against the regex and isn't added to the `pathElements` array, which is then passed through as the 'expected' path params object which raises the error. 

The regex defined [here](https://github.com/stoplightio/spectral/blob/048924d358ccb6432e368ae3d0b15cb78c72a6dd/packages/rulesets/src/oas/functions/oasPathParam.ts#L6) doesn't allow for the {*} to be defined. It is only looking for alphanumber characters and -. 

Proof that this endpoint does not pass the regex can be found [here](https://regex101.com/r/aXzwGN/1)

This can be fixed by adding * to the regex. 

**Additional context**
This error relates to fastify-swagger generating the documentation/static/{*} endpoint for some other people this may be /docs/static/{*}

Any one dealing with this error can modify their .spectral.yaml file to be. 

```yaml
extends: "spectral:oas"
overrides:
  - files:
      - "**#/paths/~1documentation~1static~1%7B*%7D"
    rules:
      path-params: "off"

```

**Comment by DanCRichards** [comment]
I have started to work on this. 

https://github.com/DanCRichards/spectral/tree/bugfix/2647-OpenAPI-Ruleset-Allow-Wildcard-In-Path

Note my latest commit doesn't take into consideration the fact that a path object should only be either alphanumeric (including -) or a wild card. 

I am looking into it but the regex that I am developing is causing failures with other test cases.

**Comment by ponelat** [comment]
There are two pieces.
One, Path Templates are a little vague, but wildcards that greedily capture whole paths aren't supported. 

Two is that names of parameters, within the curlies  `{}` [have to be percent encoded](https://github.com/OAI/OpenAPI-Specification/issues/3256#issuecomment-2173808402). I don't think this ticket is  about _naming  the  path segment *_ but if it were, then it would need to be... ` /documentation/static/{%2A}`  and have a parameter named `%2A` :D 

Going to close this out as expected behaviour. 
If folks are interested in weighing in on the Path Template, feel free to head over to https://github.com/OAI/OpenAPI-Specification

**Comment by cortexcompiler** [comment]
I have this same issue trying to us a valid [greedy path variable](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-set-up-simple-proxy.html) of `{proxy+}`. The `+` should be supported.
Form what I can tell from the [Path Templating](https://spec.openapis.org/oas/latest.html#path-templating) section in the OpenAPI Spec all unicode characters other than curly braces should be supported for template parameter names:
```
template-expression-param-name = 1*( %x00-7A / %x7C / %x7E-10FFFF ) ; every Unicode character except { and }
``` 

What I can say is that in practice I can use an OpenAPI spec file in CloudFormation to create an API Gateway REST API on AWS with a greedy path variable of `{proxy+}` and it works fine. However, the file does not pass the spectral `path-params` rule.
